File: 2012_2013_sujet_4_problemes_divers/sondage.m
function [ re] = sondage(N,n)
%UNTITLED6 Summary of this function goes here
%   Detailed explanation goes here
p=n/N;
s=sqrt(p*(1-p));
re(1)=p;
re(2)=2*(1-pnorm(abs(0.5-p)*sqrt(N)/s));

end

File: 2012_2013_sujet_4_problemes_divers/sondage.py
import numpy as np
from scipy.stats import norm


def sondage(N, n):
    """Proportion estimate and p-value for testing p=0.5."""
    p = n / N
    s = np.sqrt(p * (1 - p))
    re = np.zeros(2)
    re[0] = p
    re[1] = 2 * (1 - norm.cdf(abs(0.5 - p) * np.sqrt(N) / s))
    return re
